platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformState.java

```diff
@@ -83,56 +230,290 @@ public PlatformState copy() {
         return new PlatformState(this);
     }
 
+    /**
+     * Get the software version of the application that created this state.
+     *
+     * @return the creation version
+     */
+    public SoftwareVersion getCreationSoftwareVersion() {
+        return creationSoftwareVersion;
+    }
+
+    /**
+     * Set the software version of the application that created this state.
+     *
+     * @param creationVersion the creation version
+     */
+    public void setCreationSoftwareVersion(final SoftwareVersion creationVersion) {
+        this.creationSoftwareVersion = creationVersion;
+    }
+
     /**
      * Get the address book.
      */
+    @NonNull
     public AddressBook getAddressBook() {
-        return getChild(ChildIndices.ADDRESS_BOOK);
+        return addressBook;
     }
 
     /**
      * Set the address book.
      *
-     * @param addressBook
-     * 		an address book
+     * @param addressBook an address book
+     */
+    public void setAddressBook(@NonNull final AddressBook addressBook) {
+        this.addressBook = Objects.requireNonNull(addressBook);
+    }
+
+    /**
+     * Get the previous address book.
      */
-    public void setAddressBook(final AddressBook addressBook) {
-        setChild(ChildIndices.ADDRESS_BOOK, addressBook);
+    @Nullable
+    public AddressBook getPreviousAddressBook() {
+        return previousAddressBook;
     }
 
     /**
-     * Get the object containing miscellaneous round information.
+     * Set the previous address book.
      *
-     * @return round data
+     * @param addressBook an address book
      */
-    public PlatformData getPlatformData() {
-        return getChild(ChildIndices.PLATFORM_DATA);
+    public void setPreviousAddressBook(@Nullable final AddressBook addressBook) {
+        this.previousAddressBook = addressBook;
     }
 
     /**
-     * Set the object containing miscellaneous platform information.
+     * Get the round when this state was generated.
      *
-     * @param round
-     * 		round data
+     * @return a round number
      */
-    public void setPlatformData(final PlatformData round) {
-        setChild(ChildIndices.PLATFORM_DATA, round);
+    public long getRound() {
+        return round;
     }
 
     /**
-     * Get the previous address book.
+     * Set the round when this state was generated.
+     *
+     * @param round a round number
      */
-    public AddressBook getPreviousAddressBook() {
-        return getChild(ChildIndices.PREVIOUS_ADDRESS_BOOK);
+    public void setRound(final long round) {
+        this.round = round;
     }
 
     /**
-     * Set the previous address book.
+     * Get the running hash of all events that have been applied to this state since the beginning of time.
+     *
+     * @return a running hash of events
+     */
+    public Hash getHashEventsCons() {
```

Comment:
Is now a good time to change method names?

Reply:
Yes, changed to `getRunningEventHash()`.